sympy/sets/sets.py

```diff
@@ -1708,6 +1718,12 @@ def _contains(self, other):
         B = self.args[1]
         return And(A.contains(other), Not(B.contains(other)))
 
+    def _complement(self, other):
+        A = self.args[0]
+        B = self.args[1]
+        if other == S.UniversalSet:
+            if A == S.UniversalSet or B == S.UniversalSet:
```

Comment:
The logic you have can be stated as:

```
if other is S.UniversalSet and other in self.args:
    return self.args[1]
```

Is that right, though? Shouldn't you return the arg that is not the universal set?

```
return self.args[0] if self.args[1] is S.UniversalSet else self.args[1]
```

(Not sure about this, though.)

Reply:
If this is the case, then a test where the args are reversed should be added because nothing is failing right now.

Reply:
I see -- it is correct.  Here is a diff of the changes I would recommend:

```diff
diff --git a/sympy/sets/sets.py b/sympy/sets/sets.py
index 6ac3581..06e5023 100644
--- a/sympy/sets/sets.py
+++ b/sympy/sets/sets.py
@@ -1013,17 +1013,16 @@ def _union(self, other):
         """
         if isinstance(other, Complement):
             A, B = other.args
-            if A == S.UniversalSet:
+            if A.is_UniversalSet:
                 if Intersection(B, self).is_EmptySet:
                     return other
+                elif B.is_subset(self):
+                    return S.UniversalSet
                 else:
-                    if B.is_subset(self):
-                        return S.UniversalSet
-                    else:
-                        return Complement(S.UniversalSet, Complement(B, self))
-        if other.is_UniversalSet:
+                    return Complement(A, Complement(B, self))
+        elif other.is_UniversalSet:
             return S.UniversalSet
-        if other.is_Interval and self._is_comparable(other):
+        elif other.is_Interval and self._is_comparable(other):
             from sympy.functions.elementary.miscellaneous import Min, Max
             # Non-overlapping intervals
             end = Min(self.end, other.end)
@@ -1740,11 +1739,8 @@ def _contains(self, other):
         return And(A.contains(other), Not(B.contains(other)))

     def _complement(self, other):
-        A = self.args[0]
-        B = self.args[1]
-        if other == S.UniversalSet:
-            if A == S.UniversalSet or B == S.UniversalSet:
-                return B
+        if other.is_UniversalSet and other is self.args[0]:
+            return self.args[1]

 class EmptySet(with_metaclass(Singleton, Set)):
     """
```